### PR TITLE
[18.0][MIG] base_time_window: Migration to 18.0

### PR DESCRIPTION
**Pull request by Kimkhoi3010**

### CODE REVIEW

File: base_time_window/models/time_weekday.py
Comment (nguyenminhchien) [inline base_time_window/models/time_weekday.py:1]: remove the method name_get() . From version 17.0, we don't need to override this one any more. Check [Migration-to-version-17.0#tasks-to-do-in-the-migration](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#tasks-to-do-in-the-migration) for more information

File: base_time_window/README.rst
Comment (nguyenminhchien) [inline base_time_window/README.rst:1]: README.rst mentions to a module `test_base_window` which's available in 14.0: [test_base_window](https://github.com/OCA/server-tools/tree/14.0/test_base_time_window).

Please migrate that module to 18.0 as well.

File: base_time_window/models/time_window_mixin.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+from odoo.exceptions import ValidationError
+from odoo.tools.misc import format_time
+
+
+class TimeWindowMixin(models.AbstractModel):
+    _name = "time.window.mixin"
+    _description = "Time Window"
+    _order = "time_window_start"
+
+    # TODO patch api.constrains with field here?
```
Comment (nguyenminhchien) [inline base_time_window/models/time_window_mixin.py:19]: Handle this #TODO.

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]
Migrate `test_base_time_window` as well. Both modules in 1 PR.

File: base_time_window/models/time_weekday.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            ("1", "Tuesday"),
+            ("2", "Wednesday"),
+            ("3", "Thursday"),
+            ("4", "Friday"),
+            ("5", "Saturday"),
+            ("6", "Sunday"),
+        ],
+        required=True,
+    )
+    _sql_constraints = [("name_uniq", "UNIQUE(name)", _("Name must be unique"))]
```
Comment (nguyenminhchien) [inline base_time_window/models/time_weekday.py:23]:
```suggestion
    _sql_constraints = [("name_uniq", "UNIQUE(name)", "Name must be unique")]
```

**Review by nguyenminhchien — CHANGES_REQUESTED** [review]
- Squash these commits
image
- Create new commit `[MIG] test_base_time_window: Migration to 18.0` for migrating test_base_time_window

**Comment by nguyenminhchien** [comment]
Port or add [this commit](https://github.com/OCA/server-tools/pull/2213/commits/a6d621c611d4f32b1731683b0560290b31de70dc) to blacklist

**Comment by nilshamerlinck** [comment]
what's the status on this one?
did you have a look at `https://github.com/OCA/server-tools/pull/3030`?

**Comment by Kimkhoi3010** [comment]
Hi @nguyenminhchien @nilshamerlinck, I created a new PR [OCA#3087](https://github.com/OCA/server-tools/pull/3087) to OCA/server-tools.
(This PR was closed auto after rebased)

**Comment by Kimkhoi3010** [comment]
Hi @nguyenminhchien, @nilshamerlinck can you have take a look?
I create a new PR https://github.com/Kimkhoi3010/server-tools/pull/8

**Pull request closed without merging**